### PR TITLE
(PCP-782) Add port 8140 to master-uris if absent

### PR DESCRIPTION
**Pull request by MikaelSmith**
If master-uris includes a host with no port specified, add port 8140.
This ensures it can use the same values as Puppet's server_list and
should successfully connect to the master.

### CODE REVIEW

**Comment by MikaelSmith** [comment]
Will require a release of https://github.com/puppetlabs/leatherman/pull/258.

File: lib/src/configuration.cc [hunk 1]
```diff
@@ -841,8 +835,15 @@ void Configuration::validateAndNormalizeWebsocketSettings()
         }
     }
 
-    for (auto const& uri : master_uris_) {
-        validate_https(uri, "master-uris");
+    for (auto &uri : master_uris_) {
+        auto parsed = lth_util::uri(uri);
+        if (parsed.protocol.empty()) {
+            parsed.protocol = "https";
```
Comment (adreyer) [inline lib/src/configuration.cc]: should we reject http:// ?

Comment (MikaelSmith) [reply lib/src/configuration.cc]: Hmm, I guess curl is going to explicitly use HTTPS. I'll restore throwing if we specify something besides HTTPS.

**Comment by MikaelSmith** [comment]
Updated and running through CI.

File: lib/src/configuration.cc [hunk 2]
```diff
@@ -841,8 +835,21 @@ void Configuration::validateAndNormalizeWebsocketSettings()
...
+        }
+
+        if (parsed.protocol != "https") {
```
Comment (mruzicka) [inline lib/src/configuration.cc]:
```C++
} else if () {
```
?

Comment (MikaelSmith) [reply lib/src/configuration.cc]: Updated.

**Pull request merged**